apps/api/src/app/layouts/dtos/layout.dto.ts

```diff
@@ -0,0 +1,46 @@
+import { ApiProperty, ApiPropertyOptional } from '@nestjs/swagger';
+import {
+  ChannelTypeEnum,
+  EnvironmentId,
+  IEmailBlock,
+  ITemplateVariable,
+  OrganizationId,
+  LayoutId,
+  LayoutName,
+  UserId,
+} from '../types';
+
+export class LayoutDto {
+  @ApiPropertyOptional()
+  _id?: LayoutId;
+
+  @ApiProperty()
+  _organizationId: OrganizationId;
+
+  @ApiProperty()
+  _environmentId: EnvironmentId;
+
+  @ApiProperty()
+  _creatorId: UserId;
+
+  @ApiProperty()
+  name: LayoutName;
+
+  @ApiProperty()
+  channel: ChannelTypeEnum;
+
+  @ApiProperty()
+  content: IEmailBlock[];
+
+  @ApiProperty()
+  contentType: string;
+
+  @ApiPropertyOptional()
+  variables?: ITemplateVariable[];
+
+  @ApiProperty()
+  isDefault: boolean;
+
+  @ApiProperty()
+  isDeleted: boolean;
```

Comment:
I think isDeleted could be removed. The plugin adds 'deleted' field

Reply:
DTO is the base to return in the response so you would need it in other endpoints when returning the full entity (get layout, list layouts, etc).

Reply:
Now I understood what you meant. I got hit in a different PR by that. I am doing a mapping in the PR for the GET endpoint and will address it there. Thanks for flagging it!